src/projection.jl

```diff
@@ -255,6 +255,29 @@ end
 ##### `Base`, part II: return of the Tangent
 #####
 
+# Tuple
+function ProjectTo(xs::Tuple)
+    elements = map(xs) do x
+        x isa Union{Number, AbstractArray{<:Number}} ? ProjectTo(x) : identity
+    end
+    if all(p -> p isa ProjectTo{<:AbstractZero}, elements)
+        ProjectTo{NoTangent}()  # short-circuit if all elements project to zero
+    else
+        return ProjectTo{Tuple}(; type=typeof(xs), elements=elements)
```

Comment:
I kinda want to stick the whole primal type into the type parameter for `ProjectTo`.
We need it anyway.

This typeof is going to make accessing `project.type` type unstable, since it will make a NamedTuple with a DataType typed field.
```julia
julia> (;type = Type{typeof((1,2))}) |> typeof
NamedTuple{(:type,), Tuple{DataType}}
```

So if we want to store it as an element  we need to wrap it up in `Val`
```julia
julia> (;type = Val{typeof((1,2))}()) |> typeof
NamedTuple{(:type,), Tuple{Val{Tuple{Int64, Int64}}}}
```

Reply:
Good point. Same applies to `Ref`.